secretmanager/secretmanager_test.go

```diff
@@ -0,0 +1,344 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secretmanager
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1beta1"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"github.com/gofrs/uuid"
+	secretspb "google.golang.org/genproto/googleapis/cloud/secrets/v1beta1"
+	grpccodes "google.golang.org/grpc/codes"
+	grpcstatus "google.golang.org/grpc/status"
+)
+
+func testClient(tb testing.TB) (*secretmanager.Client, context.Context) {
+	tb.Helper()
+
+	ctx := context.Background()
+	client, err := secretmanager.NewClient(ctx)
+	if err != nil {
+		tb.Fatalf("failed to create client: %v", err)
+	}
+	return client, ctx
+}
+
+func testName(tb testing.TB) string {
+	tb.Helper()
+
+	u, err := uuid.NewV4()
+	if err != nil {
+		tb.Fatalf("failed to generate uuid: %v", err)
+	}
+	return u.String()
+}
+
+func testSecret(tb testing.TB, projectID string) (*secretspb.Secret, func()) {
+	tb.Helper()
+
+	client, ctx := testClient(tb)
+	name := testName(tb)
+
+	secret, err := client.CreateSecret(ctx, &secretspb.CreateSecretRequest{
+		Parent:   fmt.Sprintf("projects/%s", projectID),
+		SecretId: name,
+		Secret: &secretspb.Secret{
+			Replication: &secretspb.Replication{
+				Replication: &secretspb.Replication_Automatic_{
+					Automatic: &secretspb.Replication_Automatic{},
+				},
+			},
+		},
+	})
+	if err != nil {
+		tb.Fatalf("failed to create secret: %v", err)
+	}
+
+	return secret, func() { testCleanupSecret(tb, secret.Name) }
+}
+
+func testSecretVersion(tb testing.TB, parent string, payload []byte) *secretspb.SecretVersion {
+	tb.Helper()
+
+	client, ctx := testClient(tb)
+
+	version, err := client.AddSecretVersion(ctx, &secretspb.AddSecretVersionRequest{
+		Parent: parent,
+		Payload: &secretspb.SecretPayload{
+			Data: payload,
+		},
+	})
+	if err != nil {
+		tb.Fatalf("failed to create secret version: %v", err)
+	}
+	return version
+}
+
+func testCleanupSecret(tb testing.TB, name string) {
+	tb.Helper()
+
+	client, ctx := testClient(tb)
+
+	_ = client.DeleteSecret(ctx, &secretspb.DeleteSecretRequest{
+		Name: name,
+	})
+}
+
+func TestAccessSecretVersion(t *testing.T) {
+	tc := testutil.EndToEndTest(t)
+
+	payload := []byte("my-secret")
+	secret, cleanup := testSecret(t, tc.ProjectID)
+	defer cleanup()
+	version := testSecretVersion(t, secret.Name, payload)
+
+	var b bytes.Buffer
+	if err := accessSecretVersion(&b, version.Name); err != nil {
+		t.Fatal(err)
+	}
+
+	if act, exp := b.String(), string(payload); !strings.Contains(act, exp) {
+		t.Errorf("expected %q to contain %q", act, exp)
```

Comment:
Please include the function that failed. https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures